clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1813,10 +1826,16 @@ void CodeGenModule::GenOpenCLArgMetadata(llvm::Function *Fn,
 
   bool IsEsimdFunction = FD && FD->hasAttr<SYCLSimdAttr>();
 
-  if (LangOpts.SYCLIsDevice && !IsEsimdFunction)
+  if (LangOpts.SYCLIsDevice && !IsEsimdFunction) {
     Fn->setMetadata("kernel_arg_buffer_location",
                     llvm::MDNode::get(VMContext, argSYCLBufferLocationAttr));
-  else {
+
+    // The value of any "kernel_arg_runtime_aligned" metadata element is 1 for
+    // any kernel arguments that corresponds to the base pointer of an accessor
+    // and 0 otherwise.
+    Fn->setMetadata("kernel_arg_runtime_aligned",
```

Comment:
Note: If there are no kernel arguments, this metadata still gets generated with an empty list.
Included a test case for this scenario : [here](https://github.com/intel/llvm/pull/5111/files#diff-eb90775d81d9a26496cf5b517e58dd388d7e4c56ba5ecc1db8303eb3ab93030cR68)
I noticed the same behavior for `kernel_arg_buffer_location` metadata as well.

Reply:
I don't think you should generate this metadata if its empty. . I assume the existing behavior for `kernel_arg_buffer_location` is a bug. 

You can probably check the size of `argSYCLKernelRuntimeAligned` to determine if this should be generated or not

Reply:
I might be wrong here . It seems empty lists are required sometimes based on functionality. So you might want to ask the spec author.

Reply:
I honestly don't think it matters. Backends can be probably deal with both approaches. Consistency across all `kernel_arg_* metadata is probably more important.

Reply:
With my updated changes, I only generate `kernel_arg_runtime_aligned` metadata if a kernel arguments is an accessor.
This implies this metadata will not be generated if there are no kernel args.